### PR TITLE
Remove Django 1.7 and master, add Django 2.0

### PR DESCRIPTION
**Pull request by grahamu**
#### What's this PR do?

Small test matrix dependency cleanup:
* Remove all references to Django 1.7.
* Remove reference to "master".
* Add Django 2.0 explicitly to test matrix.

#### Definition of Done (check if considered and/or addressed):

- [x] Are all backwards incompatible changes documented in this PR? NO
- [x] Have all new dependencies been documented in this PR? YES
- [x] Has the appropriate documentation been updated (if applicable)? N/A
- [x] Have you written tests to prove this change works (if applicable)? N/A

### CODE REVIEW

**Comment by paltman** [comment]
https://github.com/pinax/pinax-stripe/blob/master/.circleci/config.yml Needs updating as well. Thanks!

**Comment by ticosax** [comment]
thank you for the cleanup.
master and 2.0 are two different things. I'd rather see master being kept.

**Comment by blueyed** [comment]
@grahamu 
Are you up for updating it?
There are other PRs touching the same area, but this seems to be a good common ground.

It should also remove/move py33 then.

**Comment by grahamu** [comment]
@paltman config.yml is updated and I think this is ready to go, as all tested configurations pass. Three (old) test configs show "waiting", but I have removed those from the test matrix. Not exactly sure why they are expected.

@ticosax regarding supporting "master" versus "Django 2.0", "master" often is on the bleeding edge of development, with incomplete features and bugs. This is to be expected for immature new versions. If pinax-stripe testing fails because of "master" we have to ask is the problem pinax-stripe or is Django "master" in an unstable state? This can lead to chasing a moving target as Django "master" changes. On the other hand, when a new release gets close to deployment (like v2.0 now) we can consciously add "dj20" knowing that the new version is relatively stable and a worthwhile target for compatibility.

**Comment by ticosax** [comment]
@grahamu thank you for your insights, I don't disagree. I'm used to ignore results with django master during development phase of django. And then when a release is approaching, I start to pay attention to master results, it has one major benefit, it shows the deprected path still used in your own project. Knowing that Django is very conservative, when something breaks it often means we have very old deprecated path, that worth being addressed even before the next release of django.
that said I don't want to hold this PR any longer.

It is good to go.

**Comment by blueyed** [comment]
@grahamu 
Thank you!

> Not exactly sure why they are expected.

Needs to be reconfigured in the protected branch settings of the repo.

File: .circleci/config.yml [hunk 1]
```diff
@@ -149,7 +140,7 @@ workflows:
           requires:
             - lint
       # Failing to start Django Tests for some reason
-      # - py34djmaster:
+      # - py34dj20:
       #     requires:
       #       - lint
```
Comment (blueyed) [inline .circleci/config.yml]: Maybe this is fixed by now then?

File: tox.ini [hunk 2]
```diff
@@ -40,7 +40,7 @@ deps =
     dj18: Django>=1.8,<1.9
     dj110: Django>=1.10,<1.11
     dj111: Django>=1.11a1,<2.0
-    master: https://github.com/django/django/tarball/master
+    dj20: Django<2.1
```
Comment (blueyed) [inline tox.ini]: Please keep master in the tox config.

Comment (grahamu) [reply tox.ini]: @blueyed is tox.ini used by circleci?

**Comment by grahamu** [comment]
"Protected branch settings"? What does that mean @blueyed? Is that something only repo owner can set?

**Comment by blueyed** [comment]
Yes.

File: tox.ini (hunk 2 above)
Comment (blueyed) [reply tox.ini]: Yes, but not all of it - only selected envs, via the TOXENV variable.

Comment (blueyed) [reply tox.ini]: And it is useful to still being able to run it locally, `tox -e py36-djmaster`.

Comment (grahamu) [reply tox.ini]: Ahh, _now_ why you request "master" to stay. It doesn't affect the CI builds, but is handy for local testing. Thanks @blueyed.

**Comment by grahamu** [comment]
@paltman See notes about "protected branch settings" above... I think @blueyed is saying we need to update the expected circleci status checks as it still errors on "master" which is no longer in config.yml.

**Comment by blueyed** [comment]
@grahamu 
That should only get changed once this is merged, of course - and ignored when merging it (i.e. for the status of this PR).

**Comment by blueyed** [comment]
@paltman 
Please adjust the settings for the protected branch now accordingly.

**Pull request merged**